lib/pg.ddl.js

```diff
@@ -0,0 +1,287 @@
+    `  ${domain.values.map(value => `'${value}'`).join(',\n  ')}\n);`;
+  return { type, sql };
+};
+
+const typeFromDomain = {
+  number: domain => (domain.floating ? 'double precision' : 'bigint'),
+  string: () => 'text',
+  function: () => 'text',
+  boolean: () => 'boolean',
+};
+
+const generateType = (domainName, domain) => {
+  const domainType = domain.constructor.name;
+  if (domainType === 'Object') {
+    const predefinedType = PREDEFINED_DOMAINS[domainType];
+    const type = predefinedType || typeFromDomain[domain.type](domain);
+    return { type };
+  }
+  if (domainType === 'Enum') {
+    return createEnum(domainName, domain);
+  }
+  if (domainType === 'Flags') {
+    return { type: 'bigint' };
+  }
+};
+
+const generateTypes = domains => {
+  const sqlQueries = [];
+  const types = iter(domains)
+    .filter(([domainName]) => !IGNORED_DOMAINS.includes(domainName))
+    .map(([domainName, domain]) => {
+      const { type, sql } = generateType(domainName, domain);
+      if (sql) {
+        sqlQueries.push(sql);
+      }
+      return [domainName, type];
+    })
+    .collectTo(Map);
+  return { types, typesSQL: sqlQueries.join('\n\n') };
+};
+
+const splitEntries = table => {
+  const indexes = [];
+  const links = [];
+  const properties = [];
+  iter(Object.keys(table))
+    .filter(name => table[name].constructor.name !== 'Function')
+    .each(name => {
+      const property = table[name];
+      const type = property.constructor.name;
+      const entry = { name, property };
+      if (property.index || property.unique) {
+        indexes.push(entry);
+      }
+      if (property.domain) {
+        properties.push(entry);
+      } else if (property.category) {
+        if (type !== 'Many') {
+          entry.property.required = type === 'Object' ?
+            entry.property.required :
+            type === 'Master' || type === 'Include';
+
+          properties.push(entry);
+        }
+
+        links.push(entry);
+      } else if (type === 'Unique' || type === 'Index') {
+        indexes.push(entry);
+      }
+    });
+  return { indexes, links, properties };
+};
+
+const getMaxPropLength = table =>
+  Object.keys(table).reduce(
+    (max, name) => (name.length > max ? name.length : max),
+    2
+  );
+
+const pad = (name, length) => name + ' '.repeat(length - name.length);
+
+const generateProperties = (properties, types, maxPropLength) =>
+  properties.map(({ name, property }) => {
+    let sql =
+      `${pad(name, maxPropLength)} ` +
+      (property.domain ? types.get(property.domain) : 'bigint');
+    if (property.required) {
+      sql += ' NOT NULL';
+    }
+    return sql;
+  });
+
+const generateLink = (from, to, name, type, link) => {
+  let sql =
+    `ALTER TABLE ${from} ADD CONSTRAINT fk${from}${name} ` +
+    `FOREIGN KEY (${name}) `;
+
+  sql += `REFERENCES ${link.category} (Id) `;
+  if (type === 'Object') {
+    sql += 'ON DELETE ' + (link.required ? 'RESTRICT' : 'SET NULL') + ';';
```

Comment:
You also can extract 'ON DELETE' from both branches to the statement above the `if`.